src/DeviceListener.js

```diff
@@ -79,6 +98,20 @@ export default class DeviceListener {
         this._recheck();
     }
 
+    _ensureDeviceIdsAtStartPopulated() {
+        if (this._ourDeviceIdsAtStart === null) {
+            const cli = MatrixClientPeg.get();
+            this._ourDeviceIdsAtStart = new Set(
+                cli.getStoredDevicesForUser(cli.getUserId()).map(d => d.deviceId),
+            );
+        }
+    }
+
+    _onWillUpdateDevices = async (users) => {
+        const myUserId = MatrixClientPeg.get().getUserId();
+        if (users.includes(myUserId)) this._ensureDeviceIdsAtStartPopulated();
```

Comment:
Should this event version trigger `_recheck`?

Reply:
Nope - have added a comment to hopefully clarify